File: backend/src/main/java/com/ll/domain/post/genFile/entity/PostGenFile.java
package com.ll.domain.post.genFile.entity;

import com.ll.domain.base.genFile.genFile.entity.GenFile;
import com.ll.domain.post.post.entity.Post;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@Entity
@Getter
@SuperBuilder
@NoArgsConstructor
public class PostGenFile extends GenFile {
    public enum TypeCode {
        attachment,
        thumbnail
    }

    @ManyToOne(fetch = FetchType.LAZY)
    private Post post;
    @Enumerated(EnumType.STRING)
    private TypeCode typeCode;

    @Override
    protected long getOwnerModelId() {
        return post.getId();
    }

    @Override
    protected String getTypeCodeAsStr() {
        return typeCode.name();
    }
}

File: backend/src/main/java/com/ll/domain/post/genFile/entity/PostGenFile.kt
package com.ll.domain.post.genFile.entity

import com.ll.domain.base.genFile.genFile.entity.GenFile
import com.ll.domain.post.post.entity.Post
import jakarta.persistence.*

@Entity
class PostGenFile : GenFile {
    enum class TypeCode {
        attachment,
        thumbnail
    }

    @ManyToOne(fetch = FetchType.LAZY)
    lateinit var post: Post

    @Enumerated(EnumType.STRING)
    lateinit var typeCode: TypeCode

    constructor(post: Post, typeCode: TypeCode, fileNo: Int) : super(fileNo) {
        this.post = post
        this.typeCode = typeCode
    }

    override fun getOwnerModelId(): Long {
        return post.id!!
    }

    override fun getTypeCodeAsStr(): String {
        return typeCode.name
    }
}